### PR TITLE
returning comment and message counts when the slider is shown to avoid having to actually load them all

### PR DESCRIPTION
**Pull request by 2is10**

### CODE REVIEW

File: shoebox/app/com/keepit/model/Comment.scala [hunk 1]
```diff
@@ -55,80 +57,85 @@ object Comment {
...
+  private def selectMessages[T](
+      project: RelationNode[Id[Comment],CommentEntity] => Projection[T],
+      uriId: Id[NormalizedURI],
+      userId: Id[User])(implicit conn: Connection) = {
+    val c = CommentEntity AS "c"
+    val cr = CommentRecipientEntity AS "cr"
+
+    // Get all messages *from* and/or *to* the user
+    // (side effect: User cannot be removed from own messages)
+    (SELECT (project(c)) FROM (c JOIN cr).ON("c.id = cr.comment_id") WHERE (
```
Comment (eishay) [inline shoebox/app/com/keepit/model/Comment.scala:116]: cool, didn't do such a complex query with circumflex

Comment (eishay) [inline shoebox/app/com/keepit/model/Comment.scala:116]: what is project(c)? never used it.

Comment (2is10) [inline shoebox/app/com/keepit/model/Comment.scala:116]: I factored out the projection part... callers pass it to the private helper as an anonymous function. The two callers pass different projections, but the rest of the query is the same. This ensures that the optimized count queries, which use a COUNT(c.id) projection, stay in sync with the queries that actually return the messages/comments using the simple (c.*) projection. Circumflex makes this prettier than Hibernate ever could.

Comment (eishay) [inline shoebox/app/com/keepit/model/Comment.scala:116]: Freak'n Awesome !

**Pull request merged**